### ISSUE TITLE
java class import highlight incorrectly 

### ISSUE THREAD

**Issue by liux-pro** [PrismJS/prism#3346]
**Information**
- Language: java
- Plugins: none

**Description**
If class name  all lowercase or all uppercase，it will not be marked as class correctly.

**Code snippet**

[Test page](https://prismjs.com/test.html#language=java&text=package%20com.company%3B%0A%0Aimport%20com.company2.OSS%3B%0A%0Apublic%20class%20aaa%7B%0A%20%20%20%20public%20static%20void%20main(String%5B%5D%20args)%20%7B%0A%20%20%20%20%20%20%20%20OSS.test()%3B%0A%20%20%20%20%7D%0A%7D)

<details>
<summary>The code being highlighted incorrectly.</summary>

```
package com.company;

import com.company2.OSS;

public class aaa{
    public static void main(String[] args) {
        OSS.test();
    }
}

```

</details>

**Comment by RunDevelopment** [comment]
That is because Prism highlights Java class names mostly based on Java's naming conventions. These naming conventions are important and consistently used throughout the Java ecosystem. 

Prism is neither a compiler nor a parser; we do not have access to compile information and cannot track information throughout a file. Because of these limitations, we cannot highlight all code correctly, so we typically opt for consistency instead. Consistency is very important because a syntax highlighter assigning random colors is pretty much useless. So Prism has to use approximations that result in consistent highlighting, and naming conventions are very useful in that regard.

E.g. Prism does not know that `OSS` refers to the class you imported. From naming conventions alone, `OSS` is supposed to refer to a static field. Similarly, `aaa` is supposed to refer to an instance field or variable, as class names are supposed to start with a capital letter.

To summarize: Prism is working as intended within its limitations.
